docs/howto/delete-app.txt

```diff
@@ -0,0 +1,31 @@
+==========================
+How to delete a Django app
+==========================
+
+Before deleting an application from your project you should consider that the
+app A that you are about to delete may have migrations on which app B depends
+and in case you want to remove B in the future you may also encounter an error
+like this when you want to revert migrations or running a :djadmin:`squashmigrations`:
+
+.. code-block:: shell
+
+    django.db.migrations.exceptions.NodeNotFoundError: Migration
+    <app_B>.XXXX_migration_name dependencies reference
+    nonexistent parent node ('<app_A>', 'XXXX_migration_name')
+
+Once this is said, the steps to follow are:
+
+1. Remove all references to the app (i.e imports, foreign keys).
+
+2. Remove all models from your ``models.py`` file.
+
+3. Generate your migrations running :djadmin:`makemigrations`.
+
+4. (Optional) In case other application (app B) has dependencies to the app to
```

Comment:
- This shouldn't be optional, references to a deleted app in migrations shouldn't be allowed imho 😊
- Example naming (app a, app b) is unnecessary
- Shouldn't this step be first? 🤔 Otherwise squashing will break?

Reply:
> - This shouldn't be optional, references to a deleted app in migrations shouldn't be allowed imho 😊
> - Example naming (app a, app b) is unnecessary

Agree and applied!

> - Shouldn't this step be first? 🤔 Otherwise squashing will break?

I think both ways are the same. Because if I delete the app first it will generate 2 migrations: one on the app side and other for the app where de dependecy is and the squash will result in the same.

Reply:
Thinking on it, maybe putting it in the first place would give it more importance!